File: src/main/java/grading_student/GradingStudent.java
package grading_student;

import java.util.Arrays;

public class GradingStudent {
    public void gradingStudent() {
        int[] gradeList = new int[]{73, 67, 38, 33};
        for (int i = 0; i < gradeList.length; i++) {
            if (gradeList[i] >= 38) {
                int reminder = gradeList[i] % 5;
                int least_multiple = gradeList[i] - reminder;
                int upper_multiple = least_multiple + 5;
                int difference = upper_multiple - gradeList[i];

                if (difference < 3) {
                    gradeList[i] = upper_multiple;
                }
            }
        }
        System.out.println(Arrays.toString(gradeList));
    }
}

File: src/main/kotlin/grading_student/gradingStudent.kt
package grading_student

fun gradingStudent() {

    val gradeList = mutableListOf(73, 67, 38, 33)

    for (i in gradeList.indices) {
        if (gradeList[i] >= 38) {
            val reminder = gradeList[i] % 5
            val leastMultiple = gradeList[i] - reminder
            val upperMultiple = leastMultiple + 5
            val difference = upperMultiple - gradeList[i]

            if (difference<3){
                gradeList[i] = upperMultiple
            }
        }
        println(gradeList)
    }

}